File: Tutor Demos/Week 4/Sentence.java
public class Sentence {

    private final String sentence;

    public Sentence(String sentence) {
        this.sentence = sentence;
    }

    public void display() {
        System.out.println("--------------------------------");
        System.out.println(sentence + ":");
        System.out.println("Total words in sentence: " + wordCount());
        System.out.println("Number of z words in this sentence: " + zCount());
        System.out.println(isPascalCase() ? "Sentence Is In Pascal Case" : "sentence is not in pascal case");
        System.out.println("--------------------------------");
    }

    public int wordCount() {
        return sentence.split(" ").length;
    }

    private int zCount() {
        int count = 0;
        for (String word : sentence.split(" ")) {
            if (containsZ(word)) {
                count++;
            }
        }
        return count;
    }

    private boolean containsZ(String word) {
        for (int i = 0; i < word.length(); i++) {
            if (word.toLowerCase().charAt(i) == 'z') {
                return true;
            }
        }
        return false;
    }

    //Advanced feature
    public boolean isPascalCase() {
        for (String word : sentence.split(" ")) {
            if (Character.isLowerCase(word.charAt(0))) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        String[] sentences = {"David is zhe worst tutor ever", "The Big Brown Fox Did Some Stuff I Guess", "time to muzz", "Example sentence", "exampleZ SentenzeZ"};
        for (String sentence : sentences) {
            new Sentence(sentence).display();
        }
    }
}

File: Tutor Demos/Week 4/Python/sentence.py
class Sentence:
    def __init__(self, sentence):
        self.sentence = sentence

    def display(self):
        print("--------------------------------")
        print(f"{sentence}:")
        print(f"Total words in sentence: {self.word_count()}")
        print(f"Number of z words in sentence: {self.z_count()}")
        print(f"Sentence is pascal case" if self.is_pascal_case() else "Sentence is not pascal case")
        print("--------------------------------")

    def word_count(self):
        return len(self.sentence.split(" "))

    def z_count(self):
        count = 0
        for word in self.sentence.split(" "):
            if self.contains_z(word):
                count = count + 1
        return count

    def contains_z(self, word):
        return "z" in word.lower()

    #Advanced feature
    def is_pascal_case(self):
        for word in self.sentence.split(" "):
            if word[0].islower():
                return False
        return True

if __name__ == "__main__":
    sentences = ["David is zhe worst tutor ever", "The Big Brown Fox Did Some Stuff I Guess", "time to muzz", "Example sentence", "exampleZ SentenzeZ"]
    for sentence in sentences:
        Sentence(sentence).display()